refactor: rework spear mastery and king of all weapons

LordMidas opened the pull request:
- Remove Two for One perk
- King of all Weapons reduces AP cost of thrust and prong by 1.
- King of all Weapons causes your first attack to target the body part with the lowest armor.
- Spear Mastery grants a free attack every turn with reduced damage.

Darxo commented on mod_reforged/hooks/skills/perks/perk_mastery_spear.nut:
```diff
@@ -1,8 +1,118 @@
+			id = 10,
+			type = "text",
+			icon = "ui/icons/warning.png",
+			text = ::MSU.Text.colorRed("Will expire upon switching your weapon!")
```
This is not mentioned in the perk description.

Darxo commented on mod_reforged/hooks/skills/perks/perk_mastery_spear.nut:
```diff
@@ -1,8 +1,118 @@
-	q.onAfterUpdate <- function( _properties )
+	q.m.IsSpent <- true;
+	q.m.DamageMult <- 0.75;
+	q.m.Skills <- [
```
This is a poor name choice as it can be confused with the often used `this.getContainer().m.Skills` snippet.
I suggest changing it into something like `ValidSkills`

Darxo commented on mod_reforged/hooks/skills/perks/perk_mastery_spear.nut:
```diff
@@ -1,8 +1,118 @@
+			id = 10,
+			type = "text",
+			icon = "ui/icons/warning.png",
+			text = ::MSU.Text.colorRed("Will expire upon switching your weapon!")
```
The text does not need to be all red when you already use the red warning icon. This just makes the text harder to read for little upside.

LordMidas replied:
I just followed the vanilla convention in this regard, whereby warnings are all red. And I prefer this as it makes them stand out a lot.

Darxo replied:
I can't find a single vanilla tooltip which uses `"ui/icons/warning.png"` and then proceeds to color all or any text red

Darxo commented on scripts/skills/perks/perk_rf_king_of_all_weapons.nut:
```diff
@@ -22,106 +21,72 @@ this.perk_rf_king_of_all_weapons <- ::inherit("scripts/skills/skill", {
-				this.modifyPreviewField(skill, "ActionPointCost", 0, false);
-				this.modifyPreviewField(skill, "FatigueCostMult", 1, true);
-			}
+			skill.m.ActionPointCost -= 1;
```
the `-1` should be made into a variable `this.m.ActionPointModifier` so it can be easier modified or turned off by submods or future reforged releases.

Then this line would become:
`skill.m.ActionPointCost += this.m.ActionPointModifier;`

LordMidas commented:
The PR has been changed significantly since your last review @Darxo. Based on the new design of the spear perks. So you should check the new commits. Prominent changes since your last review:
- Removed Spear Advantage perk
- Reworked through the gaps (first attack deals more armor penetration and targets body part with lower armor)
- Reworked King of all Weapons (lowers AP cost of spear attacks and the first attack every turn deals more damage)
- Refactored the skills code to use `isSkillValid` where appropriate and instead of defining "Thrust" and "Prong" manually, apply to all valid spear attacks.

LordMidas commented on mod_reforged/hooks/skills/perks/perk_mastery_spear.nut:
```diff
@@ -1,8 +1,126 @@
+			foreach (skill in this.getContainer().getSkillsByFunction((@(s) perk.isSkillValid(s))))
+			{
+				this.modifyPreviewField(skill, "ActionPointCost", 0, false);
+				this.modifyPreviewField(skill, "FatigueCostMult", 1, true);
```
I understand that these MSU functions are confusing which is why I am moving to a new more intuitive affordability preview system in MSU. These functions here are correct though, what they basically say is that during preview this skill should modify the fields of other skills by this value INSTEAD of the modification that it does in its `onAfterUpdate` function. You could read up on the [MSU affordability preview documentation](https://github.com/MSUTeam/MSU/wiki/Affordability-Preview) for detail. But honesty, once the new more intuitive system is ready then it should be much more understandable.

I will also convert this to the new MSU system now.

Darxo commented on scripts/skills/perks/perk_rf_king_of_all_weapons.nut:
```diff
@@ -1,6 +1,6 @@
-		IsSpent = false,
+		IsSpent = true,
 		ActionPointsModifier = -1,
 		DamageTotalMult = 1.25
```
comma
